packages/pagination/src/index.ts

```diff
@@ -215,3 +218,73 @@ export const createPagination = (
 
   return [paginationProps, page, setPage as Setter<number>];
 };
+
+declare module "solid-js" {
+  namespace JSX {
+    interface Directives {
+      loaderDirective: boolean
+    }
+  }
+}
+
+export type _E = JSX.Element;
+
+type loaderDirective = (ref: Element) => void
+
+/**
+ * ```typescript
+ * const [pages, asLoader, { page, setPage, setPages, end, setEnd }] = createInfiniteScroll(fetcher);
+ * ```
+ * @param fetcher (page: number) => Promise<T[]>
+ * @return `pages()` is an accessor contains array of contents 
+ * @property `pages.loading` is a boolean indicator for the loading state
+ * @property `pages.error` contains any error encountered
+ * @return `asLoader` is a directive used to set loader
+ * @method `page` is an accessor that contains page number
+ * @method `setPage` allows to manually change the page number
+ * @method `setPages` allows to manually change the contents of the page
+ * @method `end` is a boolean indicator for end of the page
+ * @method `setEnd` allows to manually change the end
+ */
+export function createInfiniteScroll<T>(
+  fetcher: (page: number) => Promise<T[]>,
+): [
+    pages: Accessor<T[]>,
+    asLoader: loaderDirective,
```

Comment:
You could just use the `Directive` type from `utils` package.
No need to make it into a separate type.
```suggestion
    asLoader: Directive,
```